test/vinyl-luatest/gh_8441_remove_root_directory_after_gc_test.lua

```diff
@@ -0,0 +1,72 @@
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all(function(cg)
+    cg.server = server:new({alias = 'master'})
+    cg.server:start()
+    cg.server:exec(function()
+        box.schema.space.create("test", {
```

Comment:
Nit: Better create the test space right in the test so that it can be repeated. (Starting the server in `before_all` is okay though.) Currently the test fails with `luatest -r 2`:

<details>

```
$ luatest -v test/vinyl-luatest/gh_8441_remove_root_directory_after_gc_test.lua
Tarantool version is 3.0.0-entrypoint-69-g14e36a526ef9
Running with --shuffle group:2424
Started on Thu Mar 16 13:00:54 2023
    vinyl-luatest.gh_8441_remove_root_directory_after_gc.test_root_directory_is_removed_after_gc ... (0.401s) Ok
=========================================================
Ran 1 tests in 0.732 seconds, 1 succeeded, 0 failed
$ luatest -v test/vinyl-luatest/gh_8441_remove_root_directory_after_gc_test.lua -r 2
Tarantool version is 3.0.0-entrypoint-69-g14e36a526ef9
Running with --shuffle group:7072
Started on Thu Mar 16 13:00:56 2023
    vinyl-luatest.gh_8441_remove_root_directory_after_gc.test_root_directory_is_removed_after_gc ... (0.400s) error
...-luatest/gh_8441_remove_root_directory_after_gc_test.lua:44: attempt to index field 'test' (a nil value)
=========================================================

Tests with errors:
------------------

1) vinyl-luatest.gh_8441_remove_root_directory_after_gc.test_root_directory_is_removed_after_gc
...-luatest/gh_8441_remove_root_directory_after_gc_test.lua:44: attempt to index field 'test' (a nil value)
stack traceback:
        ...-luatest/gh_8441_remove_root_directory_after_gc_test.lua:38: in function 'vinyl-luatest.gh_8441_remove_root_directory_after_gc.test_root_directory_is_removed_after_gc'
        ...
        [C]: in function 'xpcall'

Ran 1 tests in 0.728 seconds, 0 succeeded, 1 errored

=========================================================
Failed tests:

vinyl-luatest.gh_8441_remove_root_directory_after_gc.test_root_directory_is_removed_after_gc
```

</details>

Reply:
Done.